### PR TITLE
detect broken links in CSS-in-HTML

### PR DESCRIPTION
**Pull request by marapper**
Extracted from https://github.com/JustinBeckwith/linkinator/pull/106

Support images in `background` & `background-image` in inline style attributes.

- [ ] use ccsom or PostCSS parser (or just naive `url\([^\)]+\)` regex though?)
- ~~base-64 in `url()`~~
- ~~support `background` & `background-image` in `<style>` tags embed in HTML? (it's sometimes used as hack to lazyload images~~

### CODE REVIEW

File: src/links.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ const linksAttr = {
     'video',
   ],
   srcset: ['img', 'source'],
+  style: [''],
```
Comment (JustinBeckwith) [inline src/links.ts]: Some initial thoughts - I kind of wonder out loud how valuable parsing inline CSS really is, if it doesn't also go down the rabbit whole and parse all css files linked via `<link>`.  The problem is - that leads me down a path of thinking why not other file formats?  

I could see this spiraling and getting out of scope pretty quickly.  I'm starting to think that a separate tool that does CSS scanning specifically may be a better bet here.

Comment (marapper) [reply src/links.ts]: It's a very good point. And I totally agree about CSS' rabbit hole. 

But inline styles is a good place to stop. 

Usually only `style="background-..."` is used like img for *presentation* images. I understand that linkinator is for checking generated documentation not the sites. But for example React SSR tools can generate style-embed images too in html markup.

Comment (marapper) [reply src/links.ts]: And CSS-in-tag-style is still HTML 'part' of crawling.

**Comment by JustinBeckwith** [comment]
Thanks for the PR, but I think this is a feature we don't want to accept right now.

**Pull request closed without merging**